src/kubernetes_operator/iam_mapping.py

```diff
@@ -0,0 +1,219 @@
+        except ValueError as err:
+            if err.args[0] != "Invalid value for `conditions`, must not be `None`":
+                raise err
+    else:
+        body["metadata"]["resourceVersion"] = crds_name[crd_name]
+        extensions_api.replace_custom_resource_definition(crd_name, body)
+
+
+def full_synchronize() -> None:
+    """Synchronize all aws-auth configmap mappings with existing IamIdentityMappings."""
+    # Get Kubernetes" objects
+    configmap = API.read_namespaced_config_map("aws-auth", "kube-system")
+    identity_mappings = custom_objects_API.list_cluster_custom_object(GROUP, VERSION, PLURAL)
+
+    identities = get_identity_mappings(configmap)
+    identities = identities if isinstance(identities, list) else list()
+
+    for identity_mapping in identity_mappings["items"]:
+        identities = ensure_identity(identity_mapping["spec"], identities)
+
+    asyncio.run(apply_identity_mappings(configmap, identities))
+
+
+def get_identity_mappings(configmap: V1ConfigMap) -> list:
+    """Get the identity mappings from the aws-auth configmap as a list.
+
+    :return identities: The combined user and role mapping list
+    """
+    try:
+        identities = []
+        identities.extend(yaml.safe_load(configmap.data["mapUsers"]))
+        identities.extend(yaml.safe_load(configmap.data["mapRoles"]))
+
+        return identities
+    except yaml.YAMLError as yaml_error:
+        logger.warning("Error loading configmap mappings: %s", yaml_error)
+        return []
+
+
+async def apply_identity_mappings(existing_cm: V1ConfigMap, identity_mappings: list) -> None:
+    """Apply new identity mappings to override the existing aws-auth mapping.
+
+    :param existing_cm: The current configmap
+    :param identity_mappings: The new identity mappings
+    """
+    user_mappings = []
+    role_mappings = []
+    for identity_mapping in identity_mappings:
+        if identity_mapping.get("userarn") is not None:
+            user_mappings.append(identity_mapping)
+        elif identity_mapping.get("rolearn") is not None:
+            role_mappings.append(identity_mapping)
+        else:
+            logger.warning("Unrecognized mapping. Skipping %s", identity_mapping)
+
+    existing_cm.data["mapUsers"] = yaml.safe_dump(user_mappings)
+    existing_cm.data["mapRoles"] = yaml.safe_dump(role_mappings)
+    API.patch_namespaced_config_map("aws-auth", "kube-system", existing_cm)
+
+
+def ensure_identity(identity: dict, identity_list: list) -> list:
+    """Ensure the identity is in the list, add the identity if not present.
+
+    :param identity: The identity to check
+    :param identity_list: The list to check against
+    :return list: The updated list
+    """
+    for i, existing_identity in enumerate(identity_list):
+        # Handle existing identity
+        if existing_identity["username"] == identity["username"]:
+            identity_list[i] = identity
+            return identity_list
```

Comment:
Although it makes it cleaner, I just realized it would break updates. This method does more than add the identity. It also updates the existing one with the one passed in the params. I will add it to the doc and write a test for that case, didn't think of that.